assets/js/blocks-registry/payment-methods/payment-method-config.js

```diff
@@ -78,12 +87,24 @@ export default class PaymentMethodConfig {
 			);
 		}
 		if (
-			config.supports &&
-			typeof config.supports.savePaymentInfo !== 'undefined' &&
-			typeof config.supports.savePaymentInfo !== 'boolean'
+			typeof config.supports?.showSavedCards !== 'undefined' &&
+			typeof config.supports?.showSavedCards !== 'boolean'
+		) {
+			throw new TypeError(
+				'If the payment method includes the `supports.showSavedCards` property, it must be a boolean'
+			);
+		}
+		if ( typeof config.supports?.savePaymentInfo !== 'undefined' ) {
+			deprecated(
+				'Passing savePaymentInfo when registering a payment method is deprecated, instead you should pass showSavedCards and showSaveOption'
+			);
```

Comment:
`deprecated` will append a deprecated since... logic into your call, so you should have it in this pattern

```
	deprecated( 'Passing a callback to registerPaymentMethod()', {
			alternative: 'a config options object',
			plugin: 'woocommerce-gutenberg-products-block',
		} );
```

Since we don't intend to remove this anytime soon, no version should be supplied.

Full docs: https://github.com/WordPress/gutenberg/tree/master/packages/deprecated